docs/src/snforge-advanced-features/parametrized-testing.md

```diff
@@ -0,0 +1,134 @@
+# Parametrized Testing
+
+Sometimes, you want to run the same test logic with different inputs.
+Instead of duplicating code into separate test functions, you can use parameterized tests.
+
+Parameterized tests allow you to define multiple test cases for a single function by attaching the
+`#[test_case]` attribute.
+
+Each test case provides its own set of arguments, and `snforge` will automatically generate separate test instances for them.
+
+## Basic Example
+
+To turn a regular test into a parameterized one, add the `#[test_case(...)]` attribute above it.
+You can provide any valid Cairo expressions as arguments.
+
+Below is a simple example which checks addition of two numbers.
+
+```rust
+{{#include ../../listings/parametrized_testing_basic/tests/example.cairo}}
+```
+
+Now run:
+
+<!-- { "package_name": "parametrized_testing_basic" } -->
+```shell
+$ snforge test
+```
+
+<details>
+<summary>Output:</summary>
+
+```shell
+Collected 2 test(s) from parametrized_testing_basic package
+Running 0 test(s) from src/
+Running 2 test(s) from tests/
+[PASS] parametrized_testing_basic_integrationtest::example::test_sum_1_2_3 ([..])
+[PASS] parametrized_testing_basic_integrationtest::example::test_sum_3_4_7 ([..])
+Tests: 2 passed, 0 failed, 0 ignored, [..] filtered out
+```
+</details>
+<br>
+
+## Naming Test Cases
+
+Each parameterized test gets its own generated name. There are two ways to control it:
+
+ - **Unnamed test case** - the name is generated based on the function name and the arguments provided.
+
+    ```rust
+    #[test_case(1, 2, 3)]
+    fn test_sum(x: felt252, y: felt252, expected: felt252) {
+        assert_eq!(sum(x, y), expected);
+    }
+    ``` 
+    This will generate a test named `test_sum_1_2_3`.
```

Comment:
So saying that these are generated automatically is fine, but documenting the details of the process is problematic in a sense that this docs need to be maintained (which we will likely forget to do so).

Reply:
If you think it's necessary let's keep it, but I'm not sure if I see the benefits.

Reply:
> So saying that these are generated automatically is fine, but documenting the details of the process is problematic in a sense that this docs need to be maintained (which we will likely forget to do so).

But we're not mentioning the details of generation, are we 😅 ?